contracts/pcv/ERC4626/ERC4626PCVDeposit.sol

```diff
@@ -0,0 +1,283 @@
+        int256 startingRecordedBalance = lastRecordedBalance.toInt256();
+
+        _recordPNL(); /// update deposit amount and fire harvest event
+
+        int256 endingRecordedBalance = lastRecordedBalance.toInt256();
+
+        _updateOracle(endingRecordedBalance - startingRecordedBalance);
+
+        return lastRecordedBalance; /// return updated pcv amount
+    }
+
+    /// ------------------------------------------
+    /// ------------ Permissioned API ------------
+    /// ------------------------------------------
+
+    /// @notice withdraw tokens from the PCV allocation
+    /// non-reentrant as state changes and external calls are made
+    /// @param to the address PCV will be sent to
+    /// @param amount of tokens withdrawn
+    function withdraw(address to, uint256 amount)
+        external
+        onlyPCVController
+        globalLock(2)
+    {
+        int256 startingRecordedBalance = lastRecordedBalance.toInt256();
+
+        /// compute profit from interest accrued and emit an event
+        _recordPNL();
+
+        _withdraw(to, amount);
+
+        int256 endingRecordedBalance = lastRecordedBalance.toInt256();
+
+        _updateOracle(endingRecordedBalance - startingRecordedBalance);
+    }
+
+    /// @notice withdraw all withdrawable tokens from the vault
+    /// This is slightly different than a withdrawAll function
+    /// as the vault might not let the PCVDeposit to redeem all
+    /// its shares directly. Example if a Maple Vault does not have
+    /// enough liquidity because most of it has been borrowed
+    /// @param to the address PCV will be sent to
+    function withdrawMax(address to) external onlyPCVController globalLock(2) {
+        int256 startingRecordedBalance = lastRecordedBalance.toInt256();
+
+        /// compute profit from interest accrued and emit an event
+        _recordPNL();
+
+        /// withdraw last recorded amount as this was updated in record pnl
+        _withdraw(to, withdrawableBalance());
+
+        int256 endingRecordedBalance = lastRecordedBalance.toInt256();
+
+        _updateOracle(endingRecordedBalance - startingRecordedBalance);
+    }
+
+    /// @notice set the pcv oracle address
+    /// @param _pcvOracle new pcv oracle to reference
+    function setPCVOracle(address _pcvOracle) external override onlyGovernor {
+        address oldOracle = pcvOracle;
+        pcvOracle = _pcvOracle;
+
+        _recordPNL();
+
+        _updateOracle(lastRecordedBalance.toInt256());
+
+        emit PCVOracleUpdated(oldOracle, _pcvOracle);
+    }
+
+    /// ------------------------------------------
+    /// ------------- Helper Methods -------------
+    /// ------------------------------------------
+
+    /// @notice update the PCVOracle if the oracle is set and the updated value is not 0
+    function _updateOracle(int256 delta) private {
+        if (pcvOracle != address(0) && delta != 0) {
+            IPCVOracle(pcvOracle).updateLiquidBalance(delta);
+        }
```

Comment:
you can check if `delta != 0` and then use the `_liquidPcvOracleHook` or `_illiquidPcvOracleHook` hooks from `CoreRefV2`

https://github.com/volt-protocol/volt-protocol-core/blob/feat/prettier-folder/contracts/refs/CoreRefV2.sol#L269-L289